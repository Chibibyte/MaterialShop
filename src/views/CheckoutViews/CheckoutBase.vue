<template>
  <v-container>
    <v-row>
      <v-col cols="1" class="d-flex justify-center align-center">
        <v-icon href="icon" style="font-size: 8rem">{{ icon }}</v-icon>
      </v-col>
      <v-col>
        <v-container>
          <h1 href="title">{{ title }}</h1>
          <h2 href="subtitle">{{ subtitle }}</h2>
          <NavButton
            :routerPush="buttonFunction ? true : false"
            :runBefore="buttonFunction"
            :class="invisible ? 'd-none' : ''"
            :disabled="next_disabled"
            v-if="next_name"
            :to="next_target"
            :icon="next_icon"
            :name="next_name"
          />
        </v-container>
      </v-col>
    </v-row>
    <slot />
  </v-container>
</template>

<script>
/**
 * Business Logic
 * -display icon, title and subtitle for current CheckoutView
 * -provides NavButton with props
 *
 * Inputs:
 *    -invisible
 *    -next_disabled
 *    -title
 *    -subtitle
 *    -icon
 *    -next_target
 *    -next_icon
 *    -next_name
 * Outputs:
 *    -v-icon:href:icon
 *    -v-h1:href:title
 *    -v-h2:href:subtitle
 */
import NavButton from "@/components/NavButton.vue";
export default {
  name: "CheckoutBase",
  props: {
    buttonFunction: {
      type: Function,
      default: undefined,
    },
    invisible: {
      type: Boolean,
      default: false,
    },
    next_disabled: {
      type: Boolean,
      default: true,
    },
    title: {
      type: String,
      default: "CheckoutBase TITLE",
    },
    subtitle: {
      type: String,
      default: "CheckoutBase SUBTITLE",
    },
    icon: {
      type: String,
      default: "mdi-cart",
    },
    next_target: {
      type: String,
      default: "authenticate",
    },
    next_icon: {
      type: String,
      default: "mdi-cart",
    },
    next_name: {
      type: String,
      default: "next",
    },
  },
  components: {
    NavButton,
  },
};
</script>