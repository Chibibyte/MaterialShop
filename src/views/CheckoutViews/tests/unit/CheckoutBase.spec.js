
import Vuex from 'vuex'
import { shallowMount, createLocalVue } from '@vue/test-utils';
import CheckoutBase from '@/views/CheckoutViews/CheckoutBase.vue';

const localVue = createLocalVue();
localVue.use(Vuex);

let stubs = [
    'v-container',
    'v-row',
    'v-col',
    'v-icon'
]
let wrapper;

beforeEach(() => {
    wrapper = shallowMount(CheckoutBase, { stubs, localVue });
})

// initial
describe('Component', () => {
    test('is instantiated', () => {
        expect(wrapper.isVueInstance).toBeTruthy();
    });

    test('default snapshot', () => {
        expect(wrapper.element).toMatchSnapshot();
    })
});

describe('set output props', () => {
    test('all props set', async () => {
        await wrapper.setProps({
            icon: 'test icon', title: 'test title', subtitle: 'test subtitle',
            invisible: true, next_disabled: true, next_target: 'test next_target', next_icon: 'test next_icon', next_name: 'test next_name'
        })
        expect(wrapper.element).toMatchSnapshot();
    })

    test('all props except invisible', async () => {
        await wrapper.setProps({
            icon: 'test icon', title: 'test title', subtitle: 'test subtitle',
            next_disabled: true, next_target: 'test next_target', next_icon: 'test next_icon', next_name: 'test next_name'
        })
        expect(wrapper.element).toMatchSnapshot();
    })
})

