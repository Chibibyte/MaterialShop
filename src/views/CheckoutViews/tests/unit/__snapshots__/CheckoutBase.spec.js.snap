// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component default snapshot 1`] = `
<v-container-stub>
  <v-row-stub>
    <v-col-stub
      class="d-flex justify-center align-center"
      cols="1"
    >
      <v-icon-stub
        href="icon"
        style="font-size: 8rem;"
      >
        mdi-cart
      </v-icon-stub>
    </v-col-stub>
     
    <v-col-stub>
      <v-container-stub>
        <h1
          href="title"
        >
          CheckoutBase TITLE
        </h1>
         
        <h2
          href="subtitle"
        >
          CheckoutBase SUBTITLE
        </h2>
         
        <navbutton-stub
          class=""
          disabled="true"
          icon="mdi-cart"
          name="next"
          to="authenticate"
        />
      </v-container-stub>
    </v-col-stub>
  </v-row-stub>
   
</v-container-stub>
`;

exports[`set output props all props except invisible 1`] = `
<v-container-stub>
  <v-row-stub>
    <v-col-stub
      class="d-flex justify-center align-center"
      cols="1"
    >
      <v-icon-stub
        href="icon"
        style="font-size: 8rem;"
      >
        test icon
      </v-icon-stub>
    </v-col-stub>
     
    <v-col-stub>
      <v-container-stub>
        <h1
          href="title"
        >
          test title
        </h1>
         
        <h2
          href="subtitle"
        >
          test subtitle
        </h2>
         
        <navbutton-stub
          class=""
          disabled="true"
          icon="test next_icon"
          name="test next_name"
          to="test next_target"
        />
      </v-container-stub>
    </v-col-stub>
  </v-row-stub>
   
</v-container-stub>
`;

exports[`set output props all props set 1`] = `
<v-container-stub>
  <v-row-stub>
    <v-col-stub
      class="d-flex justify-center align-center"
      cols="1"
    >
      <v-icon-stub
        href="icon"
        style="font-size: 8rem;"
      >
        test icon
      </v-icon-stub>
    </v-col-stub>
     
    <v-col-stub>
      <v-container-stub>
        <h1
          href="title"
        >
          test title
        </h1>
         
        <h2
          href="subtitle"
        >
          test subtitle
        </h2>
         
        <navbutton-stub
          class="d-none"
          disabled="true"
          icon="test next_icon"
          name="test next_name"
          to="test next_target"
        />
      </v-container-stub>
    </v-col-stub>
  </v-row-stub>
   
</v-container-stub>
`;
